### PR TITLE
Switch to bundled electrum binary

### PR DESCRIPTION
**Pull request by alvasw**
- [Wallets: Reuse process management code from regtest setups](https://github.com/bisq-network/bisq2/commit/1d75ad9959df8149cd0f648be58877471149f3f6)
- [Handle MalformedURLException in RpcClientFactory](https://github.com/bisq-network/bisq2/commit/7591638a3146c9d32e1d9db237f28fe7094c417a)
- [Move ElectrumCli and ElectrumProcess to the main module](https://github.com/bisq-network/bisq2/commit/3cbefbd8ffd3038d5749f0b3cf7f02d0e0cfed3b)
- [AbstractRpcCliProcess: Merge run and readOutput methods](https://github.com/bisq-network/bisq2/commit/d395fd834d63cdf47f3faeb0719a3d6f3a442221)
- [ElectrumCli: Add help command](https://github.com/bisq-network/bisq2/commit/8a4de977237053591fed9f6f033fc9454b3a1c8d)
- [Electrum: Fix generated resources directory path](https://github.com/bisq-network/bisq2/commit/863bf40fd190a11ffbea8e128f3b12fd827fbb4d)
- [Implement ElectrumBinaryExtractor](https://github.com/bisq-network/bisq2/commit/237ae39c91ce5ea71e81c5b8fce9c0051480fe63)
- [Switch to bundled electrum binary](https://github.com/bisq-network/bisq2/commit/d655656bf8be104641a5c2d6e8c9ca6de4d81bf1)

### CODE REVIEW

File: wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        return Optional.empty();
+    }
+
+    private String getBinarySuffix() {
+        String osName = System.getProperty("os.name");
+        if (isRunningOnLinux()) {
+            return ElectrumBinaryExtractor.LINUX_BINARY_SUFFIX;
+        }
+        throw new UnsupportedOperationException("Unsupported OS: " + osName);
```
Comment (chimp1984) [inline wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java:126]: Is it currently only supporting Linux?

File: wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        var binaryExtractor = new ElectrumBinaryExtractor(destDirPath);
+
+        String binarySuffix = getBinarySuffix();
+        Path extractedFilePath = binaryExtractor.extractFileWithSuffix(binarySuffix);
+        binaryPath = Optional.of(extractedFilePath);
+    }
+
+    private boolean isRunningOnLinux() {
+        String osName = System.getProperty("os.name");
+        return osName.equals("Linux");
```
Comment (chimp1984) [inline wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java:77]: We have in OSUtils that feature as well...

**Review by chimp1984 — APPROVED** [review]
utACK

File: wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java (hunk 1 above)
Comment (alvasw) [reply wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java:126]: Before adding them here, I wanted to test and make sure that it runs on the other operating systems.
See: Add Electrum Windows Support  (https://github.com/bisq-network/bisq2/issues/469)

File: wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java (hunk 2 above)
Comment (alvasw) [reply wallets/electrum/src/main/java/bisq/wallets/electrum/ElectrumProcess.java:77]: I'll switch to OSUtils. Thanks!

**Pull request merged**